src/ddscxx/include/dds/core/detail/macros.hpp

```diff
@@ -119,64 +47,34 @@
 // ==========================================================================
 #endif
 
-/**
- * @internal
- * @bug OSPL-2893 SunCC template compilation issue.
- * Partial specialization to 'hardcode' the delegate required.
- */
-#ifdef __SUNPRO_CC
-#  define OSPL_2893_COMPILER_BUG
-/**
- * @internal
- * @bug OSPL-3323 Old SunCC has poor template support. SunCC 5.9/12.0 is known not to work.
- * SunCC 5.11/12.2 works. For now, let 5.10/12.1 try their luck...
- */
-#  if (__SUNPRO_CC < 0x5100)
-#    error The ISO C++ API is not supported on Solaris Studio 12.0 or earlier. Please upgrade or export INCLUDE_API_DCPS_ISOCPP=no.
-#  endif
-#endif
-
 // C++ 11 features
 // Slightly pathological - we could (for example) want to use boost traits
 // and 'STL' implementations  but compile with -std=c++11, so set a macro for
 // C++11 compile being on. This way we can always use language features
-#if defined (OSPL_DEFAULT_TO_CXX11) || defined(OSPL_USE_CXX11)
-#  define OSPL_DDS_CXX11
-#  include <cstring>
-#endif
+#define OSPL_DDS_CXX11
+#include <cstring>
 
-#if !defined(OSPL_DDS_FINAL) && defined (OSPL_DDS_CXX11)
-#  if defined (_MSC_VER) && ( _MSC_VER < 1700)
+#if defined (_MSC_VER) && ( _MSC_VER < 1700)
 // http://msdn.microsoft.com/en-us/library/vstudio/hh567368.aspx
 // 'Visual C++ in Visual Studio 2010 ... "final" was ... supported, but
 // under the different spelling "sealed". The Standard spelling and
 // semantics of "override" and "final" are now completely supported.'
-#    define OSPL_DDS_FINAL sealed
-#  else
-#    define OSPL_DDS_FINAL final
-#endif
+#  define OSPL_DDS_FINAL sealed
 #else
-#  define OSPL_DDS_FINAL
+#  define OSPL_DDS_FINAL final
 #endif
 
-#if defined(OSPL_DDS_CXX11)
-#  if defined (_MSC_VER) && (_MSC_VER <= 1800)
+#if defined (_MSC_VER) && (_MSC_VER <= 1800)
 // See: http://msdn.microsoft.com/en-us/library/vstudio/hh567368.aspx
 // "These are now supported, but with this exception: For defaulted functions,
 // the use of = default to request member-wise move constructors and move
 // assignment operators is not supported."
 // ('now' is currently VS 2013 - _MSC_VER == 1800).
-#    define OSPL_CXX11_NO_FUNCTION_DEFAULTS
-#  endif
+#  define OSPL_CXX11_NO_FUNCTION_DEFAULTS
 #endif
 
-#if defined(OSPL_DDS_CXX11)
-#  define OSPL_ENUM enum class
-#  define OSPL_ENUM_LABEL(_escope,_etype,_elabel) _escope::_etype::_elabel
-#else
-#  define OSPL_ENUM enum
-#  define OSPL_ENUM_LABEL(_escope,_etype,_elabel) _escope::_elabel
-#endif
+#define OSPL_ENUM enum class
+#define OSPL_ENUM_LABEL(_escope,_etype,_elabel) _escope::_etype::_elabel
```

Comment:
I think both OSPL_ENUM and OSPL_ENUM label can be removed altogether. They were meant to abstract the generated data model from whether the compiler supported C++11 semantics or not. Since we have now stated that C++11 support is mandatory, all data generated models no longer use this abstraction and directly specify enum classes.

Reply:
Yes, they can once the new IDL compiler is being used. Left them in for now because of the code that's generated by the older one.